geograypher/utils/indexing.py

```diff
@@ -77,3 +78,69 @@ def ensure_float_labels(query_array, full_array=None) -> (np.ndarray, dict):
     # Cast to float, since that's expected
     output_query_array = query_array.astype(float)
     return output_query_array, IDs_to_label
+
+
+def inverse_map_interpolation(
+    ijmap: np.ndarray, downsample: int = 1, fill: int = -1
+) -> np.ndarray:
+    """
+    Inverts the type of map that can be fed to skimage.transform.warp.
+
+    The basic construction is the pixel position of these maps is the position
+    in the destination image, and the value of the map is the position in the
+    source image. Therefore if location [20, 30] has value [22.2, 28.4], it
+    means that the destination image pixel [20, 30] will be sampled from the
+    source image at pixel [22.2, 28.4] (the sampler can choose to snap to the
+    closest integer value or interpolate nearby pixels).
+
+    By inverting, we seek to reverse this map through interpolation. In the
+    example above, in the output we would now have the location [22, 28]
+    have the value [20, 30] (or slightly different because it might
+    interpolate with nearby values).
+
+    Arguments:
+        ijmap ((2, H, W) numpy array): a map of the structure discussed above
+        downsample (int): Inverting (particularly griddata) can be expensive
+            for high-res image maps. You can downsample in integer steps to
+            save computation, which will downsample both the (i, j) axes. Using
+            downsample=2 will therefore interpolate on 1/4 of the pixels.
+        fill (int): Values outside of the interpolation convex hull will take
```

Comment:
This is a bit of a subtlety of relying on a map that's the same size as the image. Even though the analytic function that created the map extends beyond the bounds of the array, the inversion process doesn't have access to that information. I'm struggling to wrap my head around how important that will be in practice.

Reply:
What's the value of extending the mask if there's no data outside the image?

Reply:
Good point. I was getting hung up on the case shown below, where we render an image without distortion and then distort it to match a real one. In that case, the corners have no content. But I think this case boils down to needing to render a wider FoV if we want to have content in the corners, following distortion.

![000251-01_000283](https://github.com/user-attachments/assets/5a67a670-d575-459f-a773-b6d4c3822279)